src/Testing/src/xunit/AspNetTestInvoker.cs

```diff
@@ -59,6 +68,43 @@ protected override async Task<decimal> InvokeTestMethodAsync(object testClassIns
             return time;
         }
 
+        protected async Task<decimal> RetryAsync(RetryAttribute retryAttribute, object testClassInstance)
+        {
+            var attempts = 0;
+            var timeTaken = 0.0M;
+            for (attempts = 0; attempts < retryAttribute.MaxRetries; attempts++)
```

Comment:
Please remind me: Will this loop retry an entire `[Theory]` even if only one data set hits a problem❔

Reply:
This is a per test case/fact (at least that's the intent), I'm pretty certain of that, since there was a different extensibility point I was looking at in a different iteration of trying to implement this.